### PR TITLE
BaSP-MR-111: Applying redux to members feature

### PR DESCRIPTION
**Pull request by nicosimo8**
--**Changes:**
- Redux was applied in all of the entity methods
- Members index now just call the logic ```getMembers``` and ```deleteMember``` from redux/members/thunks.js
- ```<MemberForm />``` component now calls ```getMemberById```, ```addMember``` and ```putMember``` from redux/members/thunks.js
- If called data is in "isLoading = true" state now index shows a ```<h1>Loading</h1>```

### CODE REVIEW

File: src/Components/Members/index.js [hunk 1]
```diff
@@ -34,7 +29,10 @@ function Members(props) {
         resource={'Member'}
         onClick={() => props.history.push('/members/form')}
       />
-      <SharedTable data={members} handleDelete={deleteMemb} editLink={'members/form/'} />
+      {loading && <h1>Loading</h1>}
```
Comment (CavalloFede) [inline src/Components/Members/index.js:32]: i think we can delete the loading state and use the one we have declare in the reducer, or if we are using this one we should delete the other one

File: src/Components/Members/index.js [hunk 2]
```diff
@@ -1,30 +1,25 @@
...
-    setMembers(data.data);
-  };
+  const data = useSelector((state) => state.members);
+  const dispatch = useDispatch();
 
   useEffect(() => {
-    getMembs();
-  }, []);
+    dispatch(getMembers(setLoading));
+    setMembers(data.list);
```
Comment (CavalloFede) [inline src/Components/Members/index.js:17]: if you just call the dispatch and later pass the list to the table you don't need to use setMembers as its the same thing

File: src/Components/Members/index.js [hunk 3]
```diff
@@ -1,30 +1,25 @@
...
-      setMembers(members.filter((member) => member._id !== id));
-    } catch (error) {
-      alert(error.message);
-    }
+  const delMember = (id) => {
+    dispatch(deleteMember(id));
+    setMembers(members.filter((member) => member._id !== id));
   };
 
   return (
```
Comment (CavalloFede) [inline src/Components/Members/index.js:25]: i think we are missing the success message for the delete of a member

File: src/redux/members/thunks.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      switchModal(newMemb.error, newMemb.message);
+      dispatch(addMemberSuccess());
+    } catch (error) {
+      dispatch(addMemberError(error));
+    }
+  };
+};
+
+export const putMember = (id, member, switchModal) => {
+  console.log(member);
```
Comment (CavalloFede) [inline src/redux/members/thunks.js:76]: we can delete this console log

**Review by CavalloFede — CHANGES_REQUESTED** [review]
If you try to add a new member and dont touch the membership select it send an error due to the default value is not captured

File: src/redux/members/thunks.js (hunk 4 above)
Comment (nicosimo8) [reply src/redux/members/thunks.js:76]: Solved

File: src/Components/Members/index.js (hunk 3 above)
Comment (nicosimo8) [reply src/Components/Members/index.js:24]: Solved

File: src/Components/Members/index.js (hunk 2 above)
Comment (nicosimo8) [reply src/Components/Members/index.js]: solved

File: src/Components/Members/index.js (hunk 1 above)
Comment (nicosimo8) [reply src/Components/Members/index.js]: solved

**Comment by nicosimo8** [comment]
> It cannot be edited correctly, when a character is placed it is deleted and returns to the previous state.
> 
> pd: a console.log has been left on line 15 of members/index.js

Rdy!

File: src/redux/members/constants.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export const GET_MEMBERS_PENDING = 'GET_MEMBERS_PENDING';
+export const GET_MEMBERS_ERROR = 'GET_MEMBERS_ERROR';
+export const GET_MEMBERS_SUCCESS = 'GET_MEMBERS_SUCCESS';
+export const GET_MEMBER_BY_ID_PENDING = 'GET_MEMBER_BY_ID';
+export const GET_MEMBER_BY_ID_ERROR = 'GET_MEMBER_BY_ID_ERROR';
+export const GET_MEMBER_BY_ID_SUCCESS = 'GET_MEMBER_BY_ID_SUCCESS';
```
Comment (leo-balonchard) [inline src/redux/members/constants.js:6]: Please Replaces by:

GET_BY_ID_XXX_SUCCESS.
GET_BY_ID_XXX_PENDING
GET_BY_ID_XXX_ERROR

File: src/redux/members/reducer.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+import { PUT_MEMBER_PENDING, PUT_MEMBER_ERROR, PUT_MEMBER_SUCCESS } from './constants';
+import { DEL_MEMBER_PENDING, DEL_MEMBER_ERROR, DEL_MEMBER_SUCCESS } from './constants';
+
+const initialState = {
+  list: [],
+  item: {},
+  isLoading: false,
+  error: {}
+};
+
```
Comment (leo-balonchard) [inline src/redux/members/reducer.js:14]: Replaces by:
 const initialState = {
  list: [],
  item: {},
  isLoading: false,
  error: ' '
};

Comment (nicosimo8) [reply src/redux/members/reducer.js]: Done

File: src/redux/members/constants.js (hunk 5 above)
Comment (nicosimo8) [reply src/redux/members/constants.js]: Done

**Comment by nicosimo8** [comment]
All suggested changes done

**Review by CavalloFede — APPROVED** [review]

**Review by leo-balonchard — APPROVED** [review]
:ok_hand: :ok_hand:

**Comment by ryaev** [comment]
Everything developed is verified to be ok, a new ticket will be generated detailing low priority errors.

**Pull request merged**